client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	}
+	eak, eakN, err := e3dbClients.BoxEncryptToBase64(ak[:], encryptionKeys)
+	wrappedAccessKey := fmt.Sprintf("%s.%s", eak, eakN)
+	return wrappedAccessKey, err
+}
+
+func Contains(list []string, str string) bool {
+	for _, v := range list {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func VerifyClientCreds(credential string) (string, bool) {
+	cred := strings.Split(credential, "{")
+	if len(cred) != 2 {
+		return "incorrect format", false
+	}
+	cred = strings.Split(cred[1], "}")
+	if len(cred) != 2 {
+		return "incorrect format", false
+	}
+	cred = strings.Split(cred[0], ",")
+	var keys = make(map[string]string)
+	for _, line := range cred {
+		vals := strings.Split(line, ":")
+		key := vals[0]
+		key = strings.Split(key, "\"")[1]
+		key = strings.TrimSpace(key)
+		value := strings.Join(vals[1:], "")
+		value = strings.Split(value, "\"")[1]
+		value = strings.TrimSpace(value)
+		keys[key] = value
+	}
+	for _, reqKey := range REQUIRED_CLIENT_KEYS {
+		value, ok := keys[reqKey]
+		if ok {
+			if value == "" {
+				return fmt.Sprintf("Value for %s must be non-empty", reqKey), false
+			}
+		} else {
+			return fmt.Sprintf("Key %s must be in client credential", reqKey), false
+		}
+	}
+	matched, err := regexp.MatchString(`^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`, keys["client_id"])
+	if err != nil {
+		return fmt.Sprintf("regex error: %s occurred", err), false
+	}
+	if !matched {
+		return "Value for client_id should be in UUID format", false
+	}
+	if len(keys["public_signing_key"]) != PUBLIC_SIGNING_LENGTH {
+		return "Invalid key length: public_signing_key", false
+	}
+	if len(keys["private_signing_key"]) != PRIVATE_SIGNING_LENGTH {
+		return "Invalid key length: private_signing_key", false
+	}
+	if len(keys["public_key"]) != PUBLIC_KEY_LENGTH {
+		return "Invalid key length: public_key", false
+	}
+	if len(keys["private_key"]) != PRIVATE_KEY_LENGTH {
+		return "Invalid key length: private_key", false
+	}
+	return "", true
+}
+
+type ListedSecrets struct {
+	List      []pdsClient.ListedRecord
+	NextToken string
+}
+
+func (c *ToznySDKV3) ListSecrets(ctx context.Context, login TozIDLoginRequest, limit int, nextToken int64) (ListedSecrets, error) {
+	var sharedSecrets ListedSecrets
+
+	listRequest := storageClient.ListGroupsRequest{
+		ClientID:  uuid.MustParse(c.StorageClient.ClientID),
+		NextToken: nextToken,
+		Max:       limit,
+	}
+	responseList, err := c.StorageClient.ListGroups(ctx, listRequest)
+	if err != nil {
+		return sharedSecrets, err
+	}
+	if len(responseList.Groups) < 1 {
+		return sharedSecrets, nil
+	}
+	var found bool
+	sharedSecretList := []pdsClient.ListedRecord{}
+	for index, s := range responseList.Groups {
+		listReq := storageClient.ListGroupRecordsRequest{
+			GroupID: s.GroupID,
+		}
+		listGroupRecords, err := c.StorageClient.GetSharedWithGroup(ctx, listReq)
+		if err != nil {
+			return sharedSecrets, err
+		}
+		if len(listGroupRecords.ResultList) >= 1 {
```

Comment:
this if isn't needed,

```golang
for _, r := range listGroupRecords.ResultList {}
```
will result in zero iterations if there is nothing to iterate over